packages/remix-react/fog-of-war.ts

```diff
@@ -213,33 +221,43 @@ function getFogOfWarPaths(fogOfWar: FogOfWarInfo, router: Router) {
       return false;
     }
 
-    let matches = matchRoutes(router.routes, path, router.basename);
-    if (matches) {
-      knownGoodPaths.add(path);
-      nextPaths.delete(path);
-      return false;
-    }
```

Comment:
This could produce false positives, so we only add to `knownGoodPaths` in response to a manifest fetch where we matched on the server